arch/arm64/core/mmu.c

```diff
@@ -806,6 +806,79 @@ static uint64_t get_tcr(int el)
 	return tcr;
 }
 
+#ifdef CONFIG_DT_HAS_XEN_XEN_ENABLED
+
+static int flush_inv_mapped_pages(int (*cmo_fn)(void *, size_t),
+				  uint64_t *pte, int level, uint64_t base)
+{
+	int ret = 0;
+	uint64_t level_size = BIT(LEVEL_TO_VA_SIZE_SHIFT(level));
+	const uint64_t max_va_addr = (uint64_t)CONFIG_KERNEL_VM_BASE + CONFIG_KERNEL_VM_SIZE;
+
+	if (!xlat_use_count[table_index(pte)]) {
```

Comment:
How this can happen? `is_free_desc()` test from loop below should prevent this from ever happening.

Reply:
In this case for such table VALID bit will be 0 and `is_free_desc` will return `true` for you. On the previous level, of course.

Reply:
it is not valid bit is is use count and used only by SW

Reply:
Yes. What I am saying, is that there will no case when this `if` will fire. There is two ways to enter into `flush_inv_mapped_pages` and reach this `if`:

1. via `flush_inv_page_tbl_entries()`, in this case this `if` will check of top-level table exists. This check is a bit pointless :) 
2. via recursive call from `flush_inv_mapped_pages` itself. Recursive call is made from the loop below, after `is_free_desc` check. And if `is_free_desc` says that table entry is used, but `xlat_use_count` says that table is not used, we are in very deep trouble. 

If you want to retain this part of the code, you can replace it with 
`ASSERT(xlat_use_count[table_index(pte)]);` which is more appropriate in this case.

Reply:
ok, I'll add it and retest if it doesn't fire I delete check

Reply:
I've caught crash on board